AIAgent/ml/validation/coverage/game_managers/model/process_game_manager.py

```diff
@@ -0,0 +1,330 @@
+import json
+import logging
+import socket
+import subprocess
+from dataclasses import dataclass
+from multiprocessing.managers import Namespace
+from pathlib import Path
+from typing import Callable, Optional
+
+import torch
+from common.classes import GameFailed, GameResult, Map2Result
+from common.file_system_utils import delete_dir
+from common.game import GameMap, GameMap2SVM, GameState
+from common.network_utils import next_free_port
+from func_timeout import FunctionTimedOut
+from ml.dataset import convert_input_to_tensor
+from ml.validation.coverage.game_managers.base_game_manager import (
+    BaseGameManager,
+    BaseGamePreparator,
+)
+from ml.validation.coverage.game_managers.each_step.game_states_utils import (
+    update_game_state,
+)
+from ml.validation.coverage.game_managers.utils import set_timeout_if_needed
+from paths import CURRENT_MODEL_PATH, MODEL_KWARGS_PATH, REPORT_PATH
+from torch_geometric.data.hetero_data import HeteroData
+
+# svm substituation variables
+MAP_NAME = "MapName"
+MODEL_PATH = "ModelPath"
+STEPS_TO_PLAY = "StepsToPlay"
+DEFAULT_SEARCHER = "DefaultSearcher"
+STEPS_TO_START = "StepsToStart"
+ASSEMBLY_FULL_NAME = "AssemblyFullName"
+NAME_OF_OBJECT_TO_COVER = "NameOfObjectToCover"
+OUTPUT_DIR = "OutputDir"
+PORT = "Port"
+
+CURRENT_ONNX_MODEL_PATH = REPORT_PATH / "model.onnx"
+GAMESTATE_EXAMPLE_PATH = "../resources/onnx/reference_gamestates/7825_gameState.json"
+SVMS_OUTPUT_PATH = REPORT_PATH / "svms_output"
+
+# TODO: docs
+
+
+class ModelGamePreparator(BaseGamePreparator):
+    def __init__(
+        self,
+        namespace: Namespace,
+        model: torch.nn.Module,
+        path_to_model: str = CURRENT_MODEL_PATH,
+    ):
+        self._model = model
+        self._path_to_model = path_to_model
+        super().__init__(namespace)
+
+    def _create_onnx_model(self):
+        import_model_fqn = self._model.__class__.__module__ + ".StateModelEncoder"
+        launch_command = f"""python onyx.py --sample-gamestate {GAMESTATE_EXAMPLE_PATH}
```

Comment:
you can call it directly, import `entrypoint` as something

https://github.com/PySymGym/PySymGym/blob/5b7a6ebbd8644af9d0b18ba12b8d5da56f6f2866/AIAgent/onyx.py#L201